src/number_theory/number_field.lean

```diff
@@ -142,3 +146,179 @@ noncomputable def ring_of_integers_equiv : ring_of_integers ℚ ≃+* ℤ :=
 ring_of_integers.equiv ℤ
 
 end rat
+
+namespace adjoin_root
+
+section
+
+open_locale polynomial
+
+local attribute [-instance] algebra_rat
+
+/-- The quotient of `ℚ[X]` by the ideal generated by an irreducible polynomial of `ℚ[X]`
+is a number field. -/
+instance {f : ℚ[X]} [hf : irreducible f] : number_field (adjoin_root f) :=
+{ to_char_zero := char_zero_of_injective_algebra_map (algebra_map ℚ _).injective,
+  to_finite_dimensional := begin
+   let := (adjoin_root.power_basis (irreducible.ne_zero hf : f ≠ 0)),
+   convert power_basis.finite_dimensional this,
+   haveI : subsingleton (algebra ℚ (adjoin_root f)) := algebra_rat_subsingleton,
+   exact subsingleton.elim _ _,
+  end }
+
+end
+
+end adjoin_root
+
+namespace number_field.embeddings
+
+open set finite_dimensional complex
+open_locale complex_conjugate
+
+variables {K L : Type*} [field K] [field L]
+section number_field
+variables [number_field K] [number_field L]  (x : K)
+
+/-- The equivalence between algebra maps from a number field to `ℂ` and plain
+ring morphisms between them. -/
+noncomputable def equiv_alg : (K →ₐ[ℚ] ℂ) ≃ (K →+* ℂ) :=
+{ to_fun := coe,
+  inv_fun := λ f : K →+* ℂ, alg_hom.mk' f (λ (c : ℚ) x, map_rat_smul f _ _),
+  left_inv  := λ x, alg_hom.ext  $ by simp only [forall_const, alg_hom.coe_to_ring_hom,
+                                                 eq_self_iff_true, alg_hom.coe_mk'],
+  right_inv := λ x, ring_hom.ext $ by simp only [forall_const, alg_hom.coe_to_ring_hom,
+                                                 eq_self_iff_true, alg_hom.coe_mk'] }
```

Comment:
I suspect this only need `[char_zero K]` not `[number_field K]`

Reply:
That is correct. Should I do the proof with only ```char_zero K``` and maybe put this result another file (to be determined) since it applies to a lot more than number fields?

Reply:
Have a look near [ring_hom.to_rat_alg_hom](https://leanprover-community.github.io/mathlib_docs/algebra/algebra/basic.html#ring_hom.to_rat_alg_hom), the result can even be already there (otherwise it should go there).

Reply:
I added the result, it is now [#14772](https://github.com/leanprover-community/mathlib/tree/ring_hom.equiv_rat_alg_hom). The proof is really obvious but the result might be useful.